bundle/edu.gemini.ictd/src/main/scala/edu/gemini/ictd/dao/FeatureTablesDao.scala

```diff
@@ -0,0 +1,111 @@
+package edu.gemini.ictd.dao
+
+import edu.gemini.ictd._
+import edu.gemini.pot.sp.Instrument
+import edu.gemini.spModel.ictd.Availability
+
+import doobie.imports._
+
+import scala.collection.immutable.TreeMap
+
+import scalaz._
+import scalaz.Scalaz._
+
+import shapeless.tag.@@
+import shapeless.{ HNil, tag }
+
+/** DAO for standard instrument features like filters and gratings. */
+object FeatureTablesDao {
+
+  final case class Feature(
+    name:       String,
+    instrument: Instrument,
+    location:   Location
+  )
+
+  // Tag the various kinds of Feature to distinguish them.  They are otherwise
+  // identical triplets of (name, Instrument, Location).
+  type Circular   = Feature @@ CircularTag
```

Comment:
Are you using this like `newtype` or opaque types I guess

Reply:
Right, without the tags everything is just a `Feature` and you can easily mix them up.

Reply:
With scalaz tags it's opaque: `A @@ T` and `A` are unrelated. With shapeless `@@` (as here) it's a refinement: `A @@ T` is a subtype of `A`. Both are reasonable, depends on how you plan to use them.